### ISSUE TITLE
Granularity for Caching Components

### ISSUE THREAD

**Issue by dhargopala** [googleapis/python-aiplatform#745]
There is not direct way to specify which components should be cached in a pipeline run. the existing components will either all be cached or none of them would be cached.

For the sake of better experimentation, there should be a method/argument to explicitly specify which components to cache, without manually having to edit the compiled json.

**Comment by Ark-kun** [comment]
I agree with the idea of exposing the per-task cache reuse controls. Kubeflow Pipelines already allows that via caching option `max_cache_staleness`.

However I also think that the need to disable caching exposes deeper issues in the components.
The pipelines are supposed to be reproducible. The components are supposed to be pure. The output of a component should only depend on its arguments.

If you could describe your scenarios, we could help you find a more robust solution.

**Comment by dhargopala** [comment]
`set_caching_options` -- is a built in method, we can use this.

[Usage Example](https://github.com/kubeflow/pipelines/blob/74c7773ca40decfd0d4ed40dc93a6af591bbc190/sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_condition.py#L39)
